### PR TITLE
refactor: Add some tests for podOverrides

### PR DESCRIPTION
**Pull request by sbernauer**
# Description

*Please add a description here. This will become the commit message of the merge request later.*

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes

```[tasklist]
# Author
- [ ] Changes are OpenShift compatible
- [ ] CRD changes approved
- [ ] Integration tests passed (for non trivial changes)
```

```[tasklist]
# Reviewer
- [x] Code contains useful comments
- [ ] (Integration-)Test cases added
- [ ] Documentation added or updated
- [ ] Changelog updated
- [x] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
```

### CODE REVIEW

**Review by maltesander — COMMENTED** [review]
Just a question, LGTM otherwise.

File: src/commons/pod_overrides.rs [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+    }
+
+    #[test]
+    fn test_valid_pod_override_with_containers_and_volumes() {
+        let input = r#"
+          podOverrides:
+            spec:
+              containers:
+                - name: nifi
+                  image: docker.stackable.tech/stackable/nifi:1.23.2-stackable23.11.0
```
[the remark this answers is not in the document]
Comment (sbernauer) [reply src/commons/pod_overrides.rs:107]: I have mixed feelings :D But if your fine with it I'm also happy to discuss it at a later point of time

File: src/commons/pod_overrides.rs [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+            spec:
+              containers:
+                - image: docker.stackable.tech/stackable/nifi:1.23.2-stackable23.11.0
+        "#;
+
+        // FIXME: Ideally we would require the names of the containers to be set.  We had users using podOverrides
+        // without setting the name of the container and wondering why it didn't work.
+        serde_yaml::from_str::<Test>(input).expect("Failed to parse valid podOverride");
+    }
+}
```
[the remark this answers is not in the document]
Comment (sbernauer) [reply src/commons/pod_overrides.rs:137]: Yeah, the test can be parsed. Honestly I have not tested it, but I assume we would get a new sidecar which tries to start nifi and crashloops

[the remark this answers is not in the document]
Comment (maltesander) [reply src/commons/pod_overrides.rs:137]: is this documented somewhere?

[the remark this answers is not in the document]
Comment (sbernauer) [reply src/commons/pod_overrides.rs:137]: Nope, we only document https://docs.stackable.tech/home/stable/concepts/overrides#pod-overrides

[the remark this answers is not in the document]
Comment (maltesander) [reply src/commons/pod_overrides.rs:137]: should we explicitly state that?

[the remark this answers is not in the document]
Comment (sbernauer) [reply src/commons/pod_overrides.rs:137]: For me personally the current documentation is good enough, but happy to review any suggestions you have :)
I only know we take the PodSpec from the user and `.merge()` it onto ours. We also documented that part (including link to the ). I'm not sure what k8s does when a container has no name, but honestly I think that is on the user and not our responsibility to describe what a Pod looks like :D

**Comment by sbernauer** [comment]
Thanks for the review!

**Pull request merged**